cyvlfeat/sift/phow.py

```diff
@@ -178,35 +175,35 @@ def phow(image, verbose=False, fast=True, sizes=(4, 6, 8, 10), step=2, color='gr
                                  norm=True, fast=fast, float_descriptors=float_descriptors, verbose=verbose)
 
             temp_all_results.append(temp_results)
+        f = []
+        d = []
+        for item in temp_all_results:
+            f.append(item[0])
+            d.append(item[1])
 
-        for i in range(len(temp_all_results)):
-            f.append(temp_all_results[i][2 * i])
-            d.append(temp_all_results[i][2 * i + 1])
-
-        if color_lower == 'gray':
-            contrast = f[0][:, 2]
-        elif color_lower == 'opponent':
+        # remove low contrast descriptors note that for color descriptors the V component is thresholded
+        if color_lower in ['gray', 'opponent']:
             contrast = f[0][:, 2]
         elif color_lower == 'rgb':
             m = (f[0][:, 2], f[1][:, 2], f[2][:, 2])
-            contrast = np.mean(m, axis=1)
+            contrast = np.mean(m, axis=0)
         else:
             color_lower = 'hsv'
             contrast = f[2][:, 2]
 
-        # remove low contrast descriptors note that for color descriptors the V component is thresholded
-
         for k in range(num_channels):
-            for i, val in enumerate(contrast):
-                if val < contrast_threshold:
-                    d[k][i] = 0
-
-        dim2 = contrast.shape[0]
-        param2 = (sizes[si]) * np.ones((dim2, 1))
-        temp_frames.append(np.append(f[0], param2, axis=1))
-        frames = np.concatenate(temp_frames, axis=0)
-
-        temp_descrs.append(d[0])
-        descriptors = np.concatenate(temp_descrs, axis=0)
+            d[k][contrast < contrast_threshold] = 0
+            # for i, val in enumerate(contrast):
```

Comment:
Delete this?

Reply:
`phow` implement is converted from the MATLAB code. I rewrite some of the code to avoid redundant for-loop.